test/e2e/fixtures/when.go

```diff
@@ -258,7 +258,7 @@ func (w *When) WaitForWorkflow(options ...interface{}) *When {
 	for _, opt := range options {
 		switch v := opt.(type) {
 		case time.Duration:
-			timeout = v - 30*time.Second + defaultTimeout
+			timeout = v
```

Comment:
curious why you ended up changing this? Possibly you're right to. It does seem less kludgey and more intuitive than the old way, but I also see it was changed on purpose [here](https://github.com/argoproj/argo-workflows/commit/95bd1ccba0f79e0e00ac97988b59922f6af53bf2#diff-22b3b50dbaaa2bcad4a4d25afc8bacef81ca3eaac993d26192e8f91c761bed37). Looking at that PR, it seems like in the CI the `E2E_TIMEOUT` was specifically made longer than the default 30 seconds. So, there was some notion that all tests should be delayed by x amount if that environment variable is set. Of course, now I see the CI doesn't use that environment variable but instead uses something called [E2E_SUITE_TIMEOUT](https://github.com/argoproj/argo-workflows/blob/master/.github/workflows/ci-build.yaml#L162) cc @agilgur5 
In any case, I do see the `time.Duration` comment above isn't correct as far as "30s by default"

Reply:
I changed this because I was using different timeouts to debug my test case and I noticed it wasn't using my manually specified timeout. Subtracting 30 seconds and adding the default timeout to a manually specified value makes no sense to me.

Reply:
Here's what I'm thinking: this environment variable which used to be called `E2E_TIMEOUT` and is now called [E2E_WAIT_TIMEOUT](https://github.com/argoproj/argo-workflows/blob/master/test/e2e/fixtures/e2e_suite.go#L40) appears to be something that will allow things to run more slowly in certain environments. Is it possible to both 1. preserve the ability to run everything more slowly across the board in certain environments, and 2. have the parameters/comments be named in an intuitive way so that it's not a surprise to the caller of the function what the actual behavior is?

Reply:
(sorry, I know this wasn't the original intention of this particular issue, but I'm thinking we shouldn't remove the intention of the `E2E_WAIT_TIMEOUT` environment variable.)

Reply:
`defaultTimeout` is set [here](https://github.com/argoproj/argo-workflows/blob/8fa8f7970bfd3ccc5cff1246ea08a7771a03b8ad/test/e2e/fixtures/e2e_suite.go#L40)

```
var defaultTimeout = env.LookupEnvDurationOr("E2E_WAIT_TIMEOUT", 60*time.Second)
```

the default value for `timeout` is set to `defaultTimeout` [here](https://github.com/argoproj/argo-workflows/blob/4ffeb5f43bd1624b6b20db35f9c905b1479d4c39/test/e2e/fixtures/when.go#L251)

I think implementation in this PR is working as it should. Perhaps I'm misunderstanding your ask.

Reply:
So, I'm thinking: there are probably multiple places where `WaitForWorkflow()` is being called and some duration is being passed into it. What if someone wants to use the `E2E_WAIT_TIMEOUT` environment variable to basically add some overall buffer to **all** tests because they're running on some slow environment? (including both tests where `WaitForWorkflow()` was passed a duration, and tests where it wasn't) It seems to me that this was the original reason why it was done this way (even if it's kludgey). If somebody set `E2E_WAIT_TIMEOUT` to 2 minutes, then the tests where `WaitForWorkflow()` is not passed a duration would timeout after 2 minutes, and the tests where it is passed a duration would also include that 2 minute lag in the calculation. 

Note: I do think that including **30 seconds** in the calculation seems odd, though.

Does it make sense what I'm saying?

Reply:
I see what you're saying. Maybe something like `E2E_WAIT_TIMEOUT_BIAS` so the `defaultTimeout` could be:

```
var defaultTimeout = env.LookupEnvDurationOr("E2E_WAIT_TIMEOUT", 60*time.Second) + env.LookupEnvDurationOr("E2E_WAIT_TIMEOUT_BIAS", 0*time.Second) 
```

Is this along the lines of what you are thinking?

Reply:
yeah, something like that. Although, above where you wrote `var defaultTimeout = env.LookupEnvDurationOr("E2E_WAIT_TIMEOUT", 60*time.Second)` did you mean to write something involving "E2E_WAIT_TIMEOUT_BIAS" or `timeoutBias`?

Reply:
I edited my mistake.

Reply:
What do you think if we only have "E2E_WAIT_TIMEOUT_BIAS" and then `defaultTimeout` is set to `60*time.Second + E2E_WAIT_TIMEOUT_BIAS`?

Reply:
actually, that may not be right either - but it would be good to only have "E2E_WAIT_TIMEOUT_BIAS" if we can

Reply:
Sure, my only concern that 60 seconds is potentially too long?

Reply:
One other thing is I'm generally hesitant to change env variable names since it will be a pain for people to update.

Reply:
This line `timeout = v - 30*time.Second + defaultTimeout` only matters when a manual timeout value is specified. So it's basically saying "add the default timeout to the manually specified value" (ignoring the -30 which makes no sense). In this case  renaming the variable to include `bias` makes sense.

It makes less sense when it's used in the true default case `timeout := defaultTimeout`

Reply:
What about this small modification?:

```
var timeoutBias = env.LookupEnvDurationOr("E2E_WAIT_TIMEOUT_BIAS", 0*time.Second)
var defaultTimeout = env.LookupEnvDurationOr("E2E_WAIT_TIMEOUT", 60*time.Second) + timeoutBias
```
in the method:
```
timeout := defaultTimeout
```

And then in the switch statement, you could add the bias.
```
timeout = v + timeoutBias
```

Reply:
Yes, this.

Reply:
Thank you! Just thinking, if you see any of the tests which call this and pass a duration in intermittently fail, then it could possibly be that before they were getting an additional 30 seconds added on (E2E_WAIT_TIMEOUT=60 - 30). If so, could you bump up the duration passed in?

Reply:
Yeah, no problem.